framework/src/media/audio/audio_manager.c

```diff
@@ -584,14 +679,19 @@ audio_manager_result_t set_audio_stream_in(uint8_t channels, uint32_t sample_rat
 	}
 
 	if (channels > AUDIO_STREAM_CHANNEL_MONO) {
-		config.channels = AUDIO_STREAM_CHANNEL_STEREO;
+		channels = AUDIO_STREAM_CHANNEL_STEREO;
 	}
+
+	memset(&config, 0, sizeof(struct pcm_config));
 	config.channels = channels;
-	config.rate = sample_rate;
+	config.rate = getClosestSampRateUp(sample_rate);
```

Comment:
we should check both up & down.
some logic should added to get downsample, if there's no upsample list
Also ; remained below.

Reply:
This function returns closest sample rate. If there is no possible upsamplerate, It returns highest sample rate.